src/pages/iou/steps/MoneyRequestConfirmPage.js

```diff
@@ -72,6 +73,28 @@ function MoneyRequestConfirmPage(props) {
         [props.iou.participants, props.personalDetails],
     );
 
+    useEffect(() => {
+        const policyID = lodashGet(props.report, 'policyID', '');
+        let connectionID;
+
+        // Any money requests triggered from the FAB or from DMs/group chats won't be associated with policies.
+        // Therefore, we only want to fetch categories if we know this request is tied to an existing policy and isn't a DM/group chat.
+        if (policyID && policyID !== CONST.POLICY.POLICY_ID_FAKE) {
+            // We can't use Onyx.connect to get the policyCategories here,
+            // since we have to access the policyID from the report in Onyx.
+            // eslint-disable-next-line rulesdir/prefer-onyx-connect-in-libs
+            connectionID = Onyx.connect({
+                key: `${ONYXKEYS.POLICY_CATEGORIES}${policyID}`,
+
+                // We only want to fetch categories if:
+                // - there aren't any categories for this policy locally
+                // - the categories for this policy have been marked as stale
+                callback: (val) => (!val || val.stale) && Policy.getWorkspaceCategories(policyID),
```

Comment:
I know this sucks but I don't think there's any way to use an existing value in Onyx to key into another value with `withOnyx`. @yuwenmemon

Reply:
So this is the only way to do? @jasperhuangg

Reply:
I believe so, I can ask for a second opinion